arches_lingo/src/arches_lingo/components/generic/resource-instance-relationships/ResourceInstanceRelationshipsEditor.vue

```diff
@@ -26,27 +74,151 @@ const value = computed({
         return valRef.value;
     },
     set(value) {
-        emit("update", value);
+        const selected = options.value.filter((option) =>
+            value?.includes(option.resourceId),
+        );
+        emit(UPDATED, selected);
     },
 });
 
 const primeVuePickerVal = ref(value.value);
 watch(primeVuePickerVal, (newVal) => {
     value.value = newVal;
 });
+
+async function fetchData(page: number) {
+    try {
+        const resourceData = await fetchRelatableResources(
+            props.graphSlug,
+            props.nodeAlias,
+            page,
+        );
+        const references = resourceData.data.map(
+            (
+                resourceRecord: ResourceInstanceResult,
+            ): ResourceInstanceReference => ({
+                displayValue: resourceRecord.display_value,
+                resourceId: resourceRecord.resourceinstanceid,
+                ontologyProperty: "",
+                inverseOntologyProperty: "",
+            }),
+        );
+
+        if (page === 1) {
+            options.value = references;
+            newElements.value = resourceData.graphs.map(
+                (graphInfo: GraphInfo): NewResourceInstance => ({
+                    displayValue: $gettext("Add a new %{graphName}", {
+                        graphName: graphInfo.name,
+                    }),
+                    graphId: graphInfo.graphid,
+                }),
+            );
+        } else {
+            options.value = [...options.value, ...references];
+        }
+
+        resourceResultsPage.value = resourceData.current_page;
+        resourceResultsTotalCount.value = resourceData.total_results;
+    } catch (error) {
+        toast.add({
+            severity: ERROR,
+            life: DEFAULT_ERROR_TOAST_LIFE,
+            summary: $gettext("Failed to fetch data."),
+            detail: error instanceof Error ? error.message : undefined,
+        });
+
+        options.value = [];
+        resourceResultsPage.value = 1;
+        resourceResultsTotalCount.value = 0;
+    } finally {
+        isLoading.value = false;
+        isLoadingAdditionalResults.value = false;
+    }
+}
+
+async function onLazyLoadResources(event: VirtualScrollerLazyEvent) {
+    console.log("ohai", event);
+    if (
+        event.last >= resourceResultsPage.value * RESULTS_PER_PAGE &&
+        event.last <= resourceResultsTotalCount.value
+    ) {
+        isLoadingAdditionalResults.value = true;
+        const page = resourceResultsPage.value + 1;
+        fetchData(page);
+    }
+}
+
+function createNewResource(graphId: string) {
+    showNewResource.value = true;
+    console.log(graphId);
```

Comment:
stray log?

Reply:
Intentional - otherwise I get commit errors because graphId is unused.  I want to preserve that graphId for the next PR where I'll need it for the generic form.

Reply:
(at which point this log will go away)

Reply:
fair 👍